src/recoil/customHooks/useWidgetHandler.ts

```diff
@@ -22,4 +22,12 @@ export const useRemoveWidget = () => {
         await widgetCommunicator.removeWidgetFromDashboard(widgetId);
         set(widgetsAtom, prev => prev.filter(w => w.widgetId !== widgetId));
     }, []);
+};
+
+export const useRemoveWidgetByGraphId = () => {
+    return useRecoilCallback(({ set, snapshot }) => async (graphId: string) => {
+		const { widgetId } = snapshot.getLoadable(widgetsAtom).getValue().find(w => w.graphId === graphId);
+        await widgetCommunicator.removeWidgetFromDashboard(widgetId);
```

Comment:
can you wrap  this and the other place where we use `removeWidgetFromDashboard` in try catch ?

Reply:
sure